### DISCUSSION TITLE
Why is useSelect ignores fetchSize?

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2691] [Q&A] [answered]
### Deps:

    `"@pankod/refine-antd": "^3.53.0",`
    `"@pankod/refine-core": "^3.78.0",`

### Example:

```tsx
	const { selectProps: someDataSelectProps } = useSelect<SomeData>({
		resource: 'someResource',
		fetchSize: 5,
		optionValue: 'id',
		optionLabel: 'displayName',
		queryOptions: {
			enabled: true,
			staleTime: Timer.FifteenMinutes
		}
	})
```

This will get you the correct data back, but in the network tab you will see 2 queries being made one `without params` and another one with correct pageSize params which are -> `pageSize: 5`

ex:
```js
Req One: SomeData
Req Two: SomeData/page=0&pageSize=5
```

I'm a bit confused about why is there a fetchSize option if it fetches all data on initial render that exists anyway?

Clarification: If you use useList, with overridden pagination, it will work and make expected 1 query only
```tsx
	const res = useList<SomeData>({
		resource: 'someResource',
		config: {
			pagination: {
				current: 1,
				pageSize: 100
			},
			filters: [
				{
					field: 'sort',
					operator: 'eq',
					value: 'NameAsc'
				}
			]
		},
		queryOptions: {
			enabled: true,
			staleTime: Timer.Hour
		}
	})
```

**Comment by alicanerdurmaz** [comment, marked as answer]
Thanks for your contribution.  This bug fixed and merged. 
The process can be following here [Issue #2692](https://github.com/pankod/refine/issues/2692).
